packages/salesforce/src/Adaptor.js

```diff
@@ -659,56 +659,103 @@ function setApiVersion(apiVersion) {
 
   return version;
 }
+
 /**
- * Creates a connection.
+ * Establish a connection to Salesforce using username and password.
  * @example
- * createConnection(state)
+ * createBasicAuthConnection(state)
  * @function
  * @param {State} state - Runtime state.
  * @returns {State}
  */
-function createConnection(state) {
-  const { loginUrl, apiVersion } = state.configuration;
+async function createBasicAuthConnection(state) {
+  console.log('Attempting to connect with Basic Auth');
```

Comment:
Do we log anywhere when the connection is successful?

I would much prefer to say:

`Connected to salesforce as ${user}`

Or

`Failed to connected to salesforce as ${user}`

Ie, print the result, not the process.

When  we say "Attempting to..." it sort of shows a lack of confidence on our part, and when there's no log to follow it up it's gonna look pretty flaky.

For something routine, like logging in, we should be super confident in our logging. And when something goes wrong we should be super clear.

Reply:
I agree on with the `Connected to salesforce as ${username}` but on Failure i don't think we should log `Failed to connect as ${username}` because this suggest that the username is wrong while it could be a number of reasons. 

I would suggest we leave the error message that comes from salesforce which looks like this
```
INVALID_LOGIN: Invalid username, password, security token; or user locked out.
```
cc @josephjclark @taylordowns2000

Reply:
@mtuchi , we added the username there (if i'm remembering this correctly) because it's incredibly helpful for the end-user to know which SF user they have to debug! users _do_ get locked out, tokens become invalid, etc., etc.

printing the username is great for the salesforce admin. i don't think it will make them think that the username is wrong, i think it will help them figure out whether the password, token, or user status in salesforce is wrong.

Reply:
I don't think that error implies that username is the fault. I agree with taylor that it's a useful bit of information to report.

I also agree that the original salesforce message is fine.

We can do both!

```
console.error(`Failed to connected to salesforce as ${user}`)
console.error(e.message)  // or whatever
```

Actually, if we we throw the error, I think the runtime should report it (and if it doesn't, that's something I need to fix!)

```
console.error(`Failed to connected to salesforce as ${user}`)
throw e
````

Reply:
Okay sounds good 👍🏽 , I will use this 👇🏽
```
console.error(`Failed to connected to salesforce as ${user}`)
throw e
```

Reply:
hrmmm, i think we're not supposed to use `console.error`, right? isn't it a different stream and won't it cause timing/printout issues?

Reply:
@taylordowns2000 I was going to say: `console.error`should be fine - we redirect it to stdout anyway.

...  but then I remembered that this doesn't use our logger, so yes it will go to a different stream and cause timing problems.

I don't think that's the fault of the adaptor though. Let me just dig into something here.